cli/src/main/java/org/eclipse/hono/cli/TelemetryAndEventCli.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.qpid.proton.message.Message;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import io.vertx.proton.ProtonDelivery;
+import io.vertx.proton.ProtonHelper;
+
+/**
+ * A Client for accessing Hono's Telemetry and Event APIs to send
+ * telemetry data respectively events via the AMQP adapter.
+ */
+@Component
+@Profile("amqp-send")
+public class TelemetryAndEventCli extends AmqpCliClient {
+
+    /**
+     * The address to set on the message being sent.
+     */
+    @Value(value = "${message.address}")
+    private String messageAddress;
+
+    /**
+     * The payload to send.
+     */
+    @Value(value = "${message.payload}")
+    private String payload;
+
+    @PostConstruct
+    void start() {
+
+        final CompletableFuture<ProtonDelivery> messageSent = new CompletableFuture<>();
+
+        sendMessage(messageSent);
+
+        try {
+
+            final ProtonDelivery delivery = messageSent.join();
+            // Logs the delivery state to the console
+            printDelivery(delivery);
+        } catch (CompletionException e) {
+            LOG.info("Sending message failed [reason: {}]", e.getCause());
+        } catch (CancellationException e) {
+            // do-nothing
+        }
+        System.exit(0);
+    }
+
+    /**
+     * Connects to the AMQP adapter and send a telemetry/event message to the adapter.
+     *
+     * @param messageTracker The future to notify when the message is sent. The future is completed with the delivery
+     *            upon success or completed with an exception.
+     */
+    private void sendMessage(final CompletableFuture<ProtonDelivery> messageTracker) {
+
+        ctx.runOnContext(go -> {
+            connectToAdapter()
+            .compose(con -> {
+                adapterConnection = con;
+                return createSender();
+            })
+            .map(sender -> {
+                final Message message = ProtonHelper.message(messageAddress, payload);
+                sender.send(message, delivery -> {
+                    adapterConnection.close();
+                    messageTracker.complete(delivery);
+                });
+                return sender;
+            })
+            .otherwise(t -> {
+                messageTracker.completeExceptionally(t);
+                return null;
+            });
+        });
+     }
+
+    private void printDelivery(final ProtonDelivery delivery) {
+        final DeliveryState state = delivery.getRemoteState();
+        LOG.info("[Delivery State: {}]", state.getType());
+        switch(state.getType()) {
+        case Rejected:
+            final Rejected rejected = (Rejected) state;
+            if (rejected.getError() != null) {
+                LOG.info("[Error Condition: {}, Error Description: {}]", rejected.getError().getCondition(), rejected.getError().getDescription());
```

Comment:
please undo all the changes you made regarding the replacement of the System out PrintWriter with Logger. We deliberately chose to print to stdout because this is a command line client and its output is expected to show up on stdout and stderr.

Reply:
Sure! it was SonarLint that was complaining so i decided to use a logger for it to be happy :-) i will change it back..

Reply:
tools are now always smarter than we are, I guess ;-)